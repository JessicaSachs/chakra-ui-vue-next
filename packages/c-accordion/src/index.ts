import { h, defineComponent, PropType } from 'vue'
import { chakra, DOMElements } from '@chakra-ui/vue-system'

const CAccordion = defineComponent({
  props: {
    as: {
      type: [String] as PropType<DOMElements>,
      default: 'div',
    },
  },
  setup(props, { slots, attrs }) {
    return () => h(chakra(props.as), { ...attrs }, slots)
  },
})

export default CAccordion
