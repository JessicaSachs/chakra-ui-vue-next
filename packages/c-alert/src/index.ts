import { h, defineComponent, Component } from 'vue'

const CAlert = defineComponent({
  props: {
    as: {
      type: [String, Object],
      default: 'div',
    },
  },
  render() {
    return h(
      this.as,
      {
        ...this.$attrs,
        role: 'alert',
      },
      this.$slots.default && this.$slots.default()
    )
  },
})

export default CAlert
