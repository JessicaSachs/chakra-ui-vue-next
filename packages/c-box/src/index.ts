import { h, defineComponent, PropType } from 'vue'

const CBox = defineComponent({
  props: {
    as: {
      type: String as PropType<string>,
      default: 'div',
    },
  },
  render() {
    return h(this?.as, { ...this.$props, ...this.$attrs }, this.$slots)
  },
})

export default CBox
