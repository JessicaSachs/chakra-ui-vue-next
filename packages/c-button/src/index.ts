import { h, defineComponent, PropType } from 'vue'

const CButton = defineComponent({
  name: 'CButton',
  props: {
    as: {
      type: String as PropType<string>,
      default: 'button',
    },
  },
  render() {
    return h(
      this?.as,
      { ...this.$props, ...this.$attrs },
      this.$slots.default && this.$slots.default()
    )
  },
})

export default CButton
