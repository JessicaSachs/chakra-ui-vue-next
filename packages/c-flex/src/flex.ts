import { h, defineComponent, PropType, computed } from 'vue'
import { chakra, DOMElements } from '@chakra-ui/vue-system'

const CFlex = defineComponent({
  props: {
    as: {
      type: [Object, String] as PropType<DOMElements>,
      default: 'div',
    },
    row: {
      type: Boolean,
      default: false,
    },
    column: {
      type: Boolean,
      default: false,
    },
    reverse: {
      type: Boolean,
      default: false,
    },
  },
  setup(props, { slots, attrs }) {
    const flexDirection = computed<String>(() => {
      if (props.row && !props.reverse) return 'row'
      if (props.row && props.reverse) return 'row-reverse'
      if (props.column && !props.reverse) return 'column'
      if (props.column && props.reverse) return 'column-reverse'
      if (props.reverse) return 'row-reverse'
      return 'row'
    })
    return () =>
      h(
        chakra(props.as),
        {
          __css: {
            display: 'flex',
            flexDirection: flexDirection.value || 'row',
          },
          ...attrs,
        },
        slots
      )
  },
})

export default CFlex
