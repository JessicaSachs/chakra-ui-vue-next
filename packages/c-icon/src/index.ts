export * from './icon'
export { default as internalIcons } from './icon.internals'
export * from './icon.internals'
