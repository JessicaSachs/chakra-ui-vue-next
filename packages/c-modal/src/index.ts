export { default as CModal } from './c-modal'
