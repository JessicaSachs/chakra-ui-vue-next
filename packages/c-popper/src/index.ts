export { default as usePopper } from './vue-popper'
