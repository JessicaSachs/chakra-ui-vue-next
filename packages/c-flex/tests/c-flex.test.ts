import { CFlex } from '../src'
import { render, testA11y } from '../../test-utils/src'

const renderComponent = (props?: any) =>
  render({
    components: { CFlex },
    template: '<c-flex>Flex box !</c-flex>',
    ...props,
  })

it('should render properly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})

it('should have no a11y violations', async () => {
  await testA11y(renderComponent())
})
