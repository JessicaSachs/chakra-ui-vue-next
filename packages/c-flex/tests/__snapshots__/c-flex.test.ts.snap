// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <div
    class="css-1xhj18k"
  >
    Flex box !
  </div>
</DocumentFragment>
`;
